### PR TITLE
goto-instrument --add-cmd-line-arg: Explicitly set argv (and argc) to user-specified values

### PR DESCRIPTION
**Pull request by vmihalko**
- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
* Failed to find anything about [`--model-argc-argv`](https://github.com/diffblue/cbmc/pull/55) in `doc/cprover-manual/` so I don't know where to document `--add-cmd-line-arg`
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [x] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

This PR is related to https://github.com/diffblue/cbmc/issues/5965.

### CODE REVIEW

File: src/goto-instrument/goto_instrument_parse_options.cpp [hunk 1]
```diff
@@ -1842,7 +1870,7 @@ void goto_instrument_parse_optionst::help()
     HELP_REMOVE_CALLS_NO_BODY
     HELP_REMOVE_CONST_FUNCTION_POINTERS
     " --add-library                add models of C library functions\n"
-    " --model-argc-argv <n>        model up to <n> command line arguments\n"
+    HELP_ARGC_ARGV
```
Comment (martin-cs) [inline src/goto-instrument/goto_instrument_parse_options.cpp]: Don't worry about fixing this for this particular PR but in general we try to make sure that the code compiles after every commit.  That way we can use `git bisect` more easily.

**Review by martin-cs — APPROVED** [review]
I am fine with this but I would like a second opinion from someone who wasn't involved in https://github.com/diffblue/cbmc/issues/5965

@peterschrammel ?

**Comment by vmihalko** [comment]
I just discovered a problem. When arg string has prefix "--" or "-" (e.g. `--add-cmd-line-arg "--help"`) cbmc is interpreting it as another option, but it should be value for option...

File: regression/goto-instrument/argc-argv2/test2.desc [hunk 2]
```diff
@@ -0,0 +1,8 @@
+CORE
+main.c
+--model-argc-argv 2
+^EXIT=10$
+^SIGNAL=0$
+^VERIFICATION FAILED$
+--
+^warning: ignoring
```
Comment (TGWDB) [inline regression/goto-instrument/argc-argv2/test2.desc]: This test only tests that the `--model-argc-argv 2` is accepted, I don't think it tests the actual value `2` is used anywhere (this file fails without the flag).

File: src/config.inc [hunk 3]
```diff
@@ -76,7 +76,7 @@ endif
 OSX_IDENTITY="Developer ID Application: Daniel Kroening"
 
 # Detailed version information
-CBMC_VERSION = 5.43.0
+CBMC_VERSION = 5.44.0
```
Comment (TGWDB) [inline src/config.inc]: This should not be in this commit (but it's not important).

**Review by TGWDB — COMMENTED** [review]
I'd like to see some more tests about different kinds of inputs/cases/failures as the current tests are very limited.

Also more explanation/documentation of how these can/cannot be combined, since it appears like `--model-argc-argv` and `--add-cmd-line-arg` are mutually exclusive. Which in turns makes me wonder if someone would want to specify unknown command line arguments for some positions (e.g. I want to test my flag, but with any value rather than specifying the value).

**Comment by chrisr-diffblue** [comment]
There seems to be four commits that should not be part of this PR (commits authored by others, touching unrelated code such as the version number bump...) - I wonder if a rebase has gone wrong at some point? It would be worth a fresh rebase on top of the latest develop, and if those extra commits are still hanging around at that point, explicitly drop them so that this PR contains only your changes for this feature. That would also make it easier to review :-)

**Pull request closed without merging**